### DISCUSSION TITLE
Reading request cookies and query params in profile()

### DISCUSSION THREAD

**Discussion by nachiketAsurion** [nextauthjs/next-auth#4917] [Help] [answered]
Here is the use case  - 

User logs in to sub1.domain.com and naivgates to sub2.domain.com 

1. Sub1 sends (domain cookies with additional info and query params)=> Cognito 
2. Cognito sends Authcode to sub2.domain.com along with the domain cookies and query params
3. sub2.domain.com acquires necessary tokens based on the authcode
4. Profile is populated with the attributes received from cognito

Is there a way to access the request cookies in profile() so that, this additional info from sub1.domain.com can be added to the profile and made available as part of the session.

**Comment by ThangHuuVu** [comment, marked as answer]
Sounds like a use case for [Advance Initialization](https://next-auth.js.org/configuration/initialization#advanced-initialization) 👀 Do you think that'll help?

**Reply by nachiketAsurion** [reply]
Was able to refer cookies using this approach. Thanks!
